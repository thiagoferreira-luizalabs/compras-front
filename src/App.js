import React, { Component } from 'react';
import './App.css';
import 'semantic-ui-css/semantic.min.css';
import { Menu, Button } from 'semantic-ui-react'


class App extends Component {
  render() {
    return (
      <div className="App">
        <Menu>
          <Menu.Item header>Compras</Menu.Item>
          <Menu.Item name='aboutUs' />
          <Menu.Item name='jobs' />
          <Menu.Item name='locations' />
          <Menu.Item>
            <Button primary>Sign up</Button>
          </Menu.Item>

          <Menu.Item>
            <Button>Log-in</Button>
          </Menu.Item>
        </Menu>
      </div>
    );
  }
}

export default App;
