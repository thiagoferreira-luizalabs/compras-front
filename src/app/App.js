import React, { Component } from 'react';
import PageHeader, { ItemsMenu } from "./components/page-header/PageHeader"
import { Menus } from "./routes/Pages"
import { Menu, Segment, Sidebar } from 'semantic-ui-react'
import { Link } from "react-router-dom"

class App extends Component {

  state = { visible: false }

  handleButtonClick = () => this.setState({ visible: !this.state.visible })

  handleSidebarHide = () => this.setState({ visible: false })

  render() {
    const { isAuthenticated } = this.props.auth
    const { visible } = this.state

    return (
      <div className="app">
        {isAuthenticated && <PageHeader menus={Menus} onToggleSideBar={this.handleButtonClick} />}

        <Sidebar.Pushable as={Segment}>
          <Sidebar as={Menu} animation='overlay' icon='labeled' inverted onHide={this.handleSidebarHide} vertical visible={visible} width='thin' >
            {ItemsMenu(Menus)}
            <Menu.Item as={Link} to="/logout">Logout</Menu.Item>
          </Sidebar>

          <Sidebar.Pusher>
            <Segment basic>
              {this.props.children}
            </Segment>
          </Sidebar.Pusher>
        </Sidebar.Pushable>


      </div>
    );
  }
}

export default App;
