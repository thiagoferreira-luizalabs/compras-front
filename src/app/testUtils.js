import React from "react"
import { configure as enzymeConfigure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import sinon from "sinon";
import { MemoryRouter } from "react-router-dom";

enzymeConfigure({ adapter: new Adapter() });

const asyncAssert = (callback, done) => {
  setTimeout(() => {
    try {
      callback()
      done()
    } catch (e) {
      done.fail(e)
    }
  }, 5)
}

const createFileChangeEvent = (file) => (
  {
    target: {
      files: [file]
    }
  }
)

const MemoryRouterHOC = (children) => (<MemoryRouter>{children}</MemoryRouter>)

const toJsonSnapshot = (wrapper) => toJson(wrapper, { mode: "deep", noKey: true })

export {
  shallow, mount, sinon, MemoryRouter, toJsonSnapshot, asyncAssert, createFileChangeEvent, MemoryRouterHOC
}
