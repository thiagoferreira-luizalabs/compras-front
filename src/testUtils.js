import { configure as enzymeConfigure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import sinon from "sinon";
// import { MemoryRouter } from "react-router-dom";

enzymeConfigure({ adapter: new Adapter() });

const asyncAssert = (callback, done) => {
  setTimeout(() => {
    try {
      callback()
      done()
    } catch (e) {
      done.fail(e)
    }
  }, 5)
}

const createFileChangeEvent = (file) => (
  {
    target: {
      files: [file]
    }
  }
)

export {
  shallow, mount, sinon,  /*MemoryRouter,*/ toJson, asyncAssert, createFileChangeEvent
}
