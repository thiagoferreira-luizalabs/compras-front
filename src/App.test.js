import React from 'react';
import App from './App';
import { mount, toJson } from "./testUtils"

it("Renders without crashing", () => {
  let wrapper = mount(<App />)
  expect(toJson(wrapper)).toMatchSnapshot()
})