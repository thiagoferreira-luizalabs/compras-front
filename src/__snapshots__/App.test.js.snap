// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without crashing 1`] = `
<App>
  <div
    className="App"
  >
    <Menu>
      <div
        className="ui menu"
      >
        <MenuItem
          header={true}
        >
          <div
            className="header item"
            onClick={[Function]}
          >
            Compras
          </div>
        </MenuItem>
        <MenuItem
          name="aboutUs"
        >
          <div
            className="item"
            onClick={[Function]}
          >
            About Us
          </div>
        </MenuItem>
        <MenuItem
          name="jobs"
        >
          <div
            className="item"
            onClick={[Function]}
          >
            Jobs
          </div>
        </MenuItem>
        <MenuItem
          name="locations"
        >
          <div
            className="item"
            onClick={[Function]}
          >
            Locations
          </div>
        </MenuItem>
        <MenuItem>
          <div
            className="item"
            onClick={[Function]}
          >
            <Button
              as="button"
              primary={true}
              role="button"
            >
              <button
                className="ui primary button"
                onClick={[Function]}
                role="button"
              >
                Sign up
              </button>
            </Button>
          </div>
        </MenuItem>
        <MenuItem>
          <div
            className="item"
            onClick={[Function]}
          >
            <Button
              as="button"
              role="button"
            >
              <button
                className="ui button"
                onClick={[Function]}
                role="button"
              >
                Log-in
              </button>
            </Button>
          </div>
        </MenuItem>
      </div>
    </Menu>
  </div>
</App>
`;
